scripts/scil_connectivity_compute_pca.py

```diff
@@ -150,179 +192,117 @@ def autolabel(rects, axs):
 
     with open(args.list_ids) as f:
         subjects = f.read().split()
 
     if args.input_connectoflow:
         # Loading all matrix.
         logging.info('Loading all matrices from a Connectoflow output...')
-        dictionary = {m: [load_matrix_in_any_format(f'{args.in_folder}/{a}/Compute_Connectivity/{m}.npy')
-                          for a in subjects]
-                      for m in args.metrics}
+        files_per_metric = [[os.path.join(
+            args.in_folder, a, 'Compute_Connectivity', '{}.npy'.format(m))
+            for a in subjects]
+            for m in args.metrics]
     else:
         logging.info('Loading all matrices...')
-        dictionary = {m: [load_matrix_in_any_format(f'{args.in_folder}/{a}_{m}.npy') for a in subjects]
-                      for m in args.metrics}
-        # Assert that all metrics have the same number of subjects.
-        nb_sub = [len(dictionary[m]) for m in args.metrics]
-        assert all(x == len(subjects) for x in nb_sub), "Error, the number of matrices for each metric doesn't match" \
-                                                        "the number of subject in the id list." \
-                                                        "Please validate input folder."
+        files_per_metric = [[os.path.join(
+            args.in_folder, '{}_{}.npy'.format(a, m))
+            for a in subjects]
+            for m in args.metrics]
+
+    assert_inputs_exist(parser, sum(files_per_metric, []))
+    matrices_per_metric = [[load_matrix_in_any_format(f) for f in files]
+                           for files in files_per_metric]
 
     # Setting individual matrix shape.
-    mat_shape = dictionary[args.metrics[0]][0].shape
+    mat_shape = matrices_per_metric[0][0].shape
 
-    if args.not_only_common:
+    if args.all_edges:
         # Creating input structure using all edges from all subjects.
         logging.info('Creating PCA input structure with all edges...')
-        df = generate_pca_input(dictionary)
     else:
-        m1 = extracting_common_cnx(dictionary, 0)
-        m2 = extracting_common_cnx(dictionary, 1)
-
-        if m1.shape != mat_shape:
-            parser.error("Extracted binary mask doesn't match the shape of individual input matrix.")
-
-        if np.sum(m1) != np.sum(m2):
-            parser.error("Different binary mask have been generated from 2 different metrics, \n "
-                         "please validate input data.")
-        else:
-            logging.info('Data shows {} common connections across the population.'.format(np.sum(m1)))
-
-        dictionary = apply_binary_mask(dictionary, m1)
-
-        # Creating input structure.
         logging.info('Creating PCA input structure with common edges...')
-        df = generate_pca_input(dictionary)
 
-    # Setting 0 values to nan.
-    df[df == 0] = 'nan'
-
-    # Standardized the data.
+        common_edges_mask = np.sum(matrices_per_metric[0]) != 0
```

Comment:
I'm unsure about this; the common edges are supposed to be across all input subjects. Using the `--all_edges` flag returns the same results as if we don't use it.